test-files/golden-tests/config/sfinae/redeclare.cpp

```diff
@@ -1,6 +1,28 @@
 // issue #850
 
-#include <type_traits>
+namespace std
+{
+template <bool C, typename T = void>
+struct enable_if
+{
+    using type = T;
+};
+template <typename T>
+struct enable_if<false, T>
+{};
+
+template <bool C, typename T = void>
+using enable_if_t = typename enable_if<C, T>::type;
+
+template <typename T>
+struct is_class
+{
+    static constexpr bool value = true;
+};
+
+template <typename T>
+bool is_class_v = is_class<T>::value;
+}
```

Comment:
Can we add these to some headers instead, so that we can just reuse them across tests? Ditto for other instances.

Reply:
added common header in an implicit include directory, is this what you had in mind?

Reply:
Yes, thanks!